extmod/modbtree.c

```diff
@@ -246,8 +246,10 @@ STATIC mp_obj_t btree_iternext(mp_obj_t self_in) {
     }
 }
 
+#pragma GCC diagnostic ignored "-Wunused-parameter"
```

Comment:
Why is this needed? Does it apply to the whole file? I've done `(void) var;` instead. That is clearer about which parameter is unused.

Reply:
This is leftover from the `_subscr` changes - removing these.